prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -385,7 +385,12 @@ void rxr_pkt_handle_rma_read_completion(struct rxr_ep *ep,
 		if (read_entry->x_entry_type == RXR_TX_ENTRY) {
 			tx_entry = ofi_bufpool_get_ibuf(ep->tx_entry_pool, read_entry->x_entry_id);
 			assert(tx_entry && tx_entry->cq_entry.flags & FI_READ);
-			rxr_cq_write_tx_completion(ep, tx_entry);
+			if (tx_entry->fi_flags & FI_COMPLETION) {
```

Comment:
We seem to be doing this check anytime we call `rxr_cq_write_tx_completion`, can we move the selective completion logic inside it? We have the counter logic in there too.

Reply:
Good idea. will do